unlock-app/src/middlewares/storageMiddleware.js

```diff
@@ -0,0 +1,39 @@
+import {
+  CREATE_LOCK,
+  UPDATE_LOCK,
+  ADD_LOCK,
+  LOCK_DEPLOYED,
+} from '../actions/lock'
+import { setLockName } from '../actions/storage'
+import StorageService from '../services/storageService'
+
+export default function storageMiddleware({ dispatch }) {
+  const storageService = new StorageService()
+
+  return function(next) {
+    return function(action) {
+      switch (action.type) {
+      case CREATE_LOCK:
+        storageService.storeLockDetails(action.lock)
+        break
+      case UPDATE_LOCK:
+        storageService.lockLookUp(action.address).then(results => {
+          dispatch(setLockName(action.address, results.data.name))
+        })
+        break
+      case ADD_LOCK:
```

Comment:
I wonder if we should maybe move the `UPDATE_LOCK` processing down here. Since we just need the address and the `ADD_LOCK` action has it, maybe we can invoke `lockLookUp` here?